src/catalog/index_catalog.cpp

```diff
@@ -270,6 +283,28 @@ std::shared_ptr<IndexCatalogObject> IndexCatalog::GetIndexObject(
   return nullptr;
 }
 
+std::unordered_map<oid_t, std::shared_ptr<IndexCatalogObject>>
+IndexCatalog::GetIndexObjects(concurrency::TransactionContext *txn) {
+  std::unordered_map<oid_t, std::shared_ptr<IndexCatalogObject>> result_indexes;
+  if (txn == nullptr) {
+    throw CatalogException("Transaction is invalid!");
+  }
+  // try get from cache
+  auto pg_table = Catalog::GetInstance()
+                      ->GetSystemCatalogs(database_oid)
+                      ->GetTableCatalog();
```

Comment:
This is not the correct way to do this. We should not directly access the catalog instance. We should get the database catalog object from `txn->catalog_cache_`, then get table objects, and then index objects. Everything should go through the local catalog cache of the transaction but no the global catalog instance.